### ISSUE TITLE
Unpaid Earnings not incrementing

### ISSUE THREAD

**Issue by amdrew** [AffiliateWP/AffiliateWP#2189]
There's been 3+ reports from customers whereby the unpaid earnings does not increase.

It seems this affects both the `Unpaid Earnings` on the `Statistics` tab of the Affiliate Area, and the `Unpaid Earnings` column in the admin (they both use the same `affwp_get_affiliate_unpaid_earnings()` function).

All customers to my knowledge are running AffiliateWP 2.0.6 and one stated that this started happening after they upgraded from a pre 2.0 release. Recounting an affiliate's unpaid earnings in the admin fixed the issue for a customer I was dealing with however doing this for every affiliate is not ideal.

I know @gspice spotted the same issue on a customer's site. So far I've been unsuccessful in replicating the issue.

Related ticket/s:
https://secure.helpscout.net/conversation/353637597/57873?folderId=205529

**Comment by gspice** [comment]
Here is the ticket Andrew spoke of that I initially answered:

https://secure.helpscout.net/conversation/352780596/57745/?folderId=1170842

I logged in to her site to use the Recount Affiliate Stats tool for Unpaid but it didn't correct the issue.   Rami replied and asked her to update all of her plugins and WordPress versions (WP was 4.6)  I will circle back with her tomorrow to check the status.

**Comment by gspice** [comment]
Update on the above user ticket - she wrote back saying now her `Paid Earnings` and `Visits` 
 counts are off on these same two affiliates with the first issue.  I logged on, went back to Recount Affiliate Stats and this time I entered the specific usernames and selected Unpaid Earnings to Recount.  This has worked for Unpaid Earnings - now the affiliates page shows the correct dollar amount in the `Unpaid Earnings` column.  Unfortunately the same steps did not correct the `Paid Earnings` and `Visits` amounts for these two affiliates. 

![paid-earnings-visits-off](https://cloud.githubusercontent.com/assets/3018245/25450067/a420fc40-2a72-11e7-862b-29305a27cf2f.png)

**Comment by DrewAPicture** [comment]
It would be helpful to have @pippinsplugins or @ramiabraham step through the unpaid earnings code just for a fresh set of eyes. They might see something that I missed and/or something we should be (and currently aren't) testing for.

**Comment by pippinsplugins** [comment]
Did a quick scan and it all _looks_ like it should work and is nearly identical to how it's been for a long time. I've also stepped through some manual tests and both paid and unpaid earnings were updated properly in all cases for me.

**Comment by pippinsplugins** [comment]
This has been fixed now so am going to mark as fixed. Re-open if you see it again.

**Comment by RobertoGongora** [comment]
I'm having this issue as well, I had the 2.0.6 version too and I upgraded to the 2.1.13 recently to see if it fixed it but it didn't. I made sure the database upgrade process was completed, I've tried using the Recount tool but nothing seems to work. 

As a "side-effect" of sorts, if I try sorting the Affiliates table by Unpaid Earnings, it breaks and says there are no affiliates. 

Currently my table looks like this: ![Table](https://cdn.discordapp.com/attachments/272887645791584270/413380616445231104/unknown.png)

I've tried debugging the code, and I found that the "unpaid_earnings" field of the Affiliate Class does nothing, the `affwp_get_affiliate_unpaid_earnings` function just returns the `$affiliate->unpaid_earnings;` which is not defined.

**Comment by RobertoGongora** [comment]
I figured out what it might be. In my `wp_affiliate_wp_affiliates` table there is no `unpaid_earnings` column. I'm not sure why, or how to fix this other than manually adding it. Is there any tool I can run to force a DB upgrade? I'm not sure why the field isn't there.

**Comment by amdrew** [comment]
@RobertoGongora Hmm, can you try deactivating AffiliateWP and then reactivating it again? That "should" recreate the unpaid_earning column.

**Comment by amdrew** [comment]
@RobertoGongora Did you manage to get it sorted? If you're still not able to generate the column please reach out to us at https://affiliatewp.com/support/ and we'll help you resolve this.

Another thought - If after you have regenerated your unpaid earnings column, you may need to force an upgrade routine to recount the unpaid earnings. You'll need to do this if all your totals in the unpaid earnings column on the affiliates page are $0.00. 

To do this, delete the `affwp_completed_upgrades` option row from `wp_options`. Then, refresh the WP admin and you'll see an upgrade notice from AffiliateWP. Run the upgrade and all the totals should then be calculated correctly.

**Comment by RobertoGongora** [comment]
Hey, sorry, yes I was able to fix the issue.

I did what you suggested, removed the completed upgrades option and forced
the upgrade via the WP admin but it didn’t help the issue.

What I ended up doing was manually adding the row to my table using raw SQL
queries and then I ran the unpaid earnings recount tool and everything
worked.

What worries me about this method is that I’m not sure if there are
additional modifications to the database that I’m not aware of and may be
still missing :/
